firmware.py

```diff
@@ -63,11 +69,63 @@ def details(company_id):
     )
     records = companyCursor.fetchmany(1)
     if len(records) == 0:
+        app.config['MY_MESSAGE']="The requested company was not found!"
         return redirect(url_for('not_found'))
-
+    categoryCursors = db.execute( 'select category_type from category where category_id="%s"' % records[0]['category_id'])
+    category = categoryCursors.fetchall()
     reviewsCursor = db.execute(
         'select user, review from reviews where company_id="%s" order by id desc'
         % company_id
     )
     reviews = reviewsCursor.fetchall()
-    return render_template('details.html', reviews=reviews, company=records[0])
+    return render_template('details.html', reviews=reviews, company=records[0], category=category[0]['category_type'])
+
+
+@app.route('/add/company', methods=['GET', 'POST'])
+def add_company():
+    db = get_db()
+    submitted_data = request;
+    if request.method == 'POST':
+        submitted_data = request
+        a = request.form['company_name']
+        if(a == ""):
+            flash("Please add a name for your company!")
```

Comment:
Do not use flash messages for errors. Use a dict with field names as keys and pass it to the render method.